Snake/BusinessLayer/maze.cs

```diff
@@ -8,27 +8,39 @@ namespace BusinessLayer
 {
      public class maze
     {
-        private const int border = 1;
-        private const int no_border = 0;
+        private const int BORDER = 1;
+        private const int NO_BORDER = 0;
         private int width, height;
-
+        private int[,] Maze;
+        Random random_number;
 
 
 
         public maze(int w,int h)
         {
             width = w;
             height = h;
+            Maze = new int[height, width];
+            random_number = new Random();
         }
 
         public int[,] CreateMaze()
         {
-            int[,] maze = new int[height,width];
+            Generate_Basic_Maze();
+
+            Generate_All_Obstacles();
+            
+            return Maze;
+        }
+
+
+        public void Generate_Basic_Maze()
+        {
             for (int i = 0; i < height; i++)
```

Comment:
consider using linq instead of these loops. it is not faster but it is easier to read (and write once you know the syntax)